src/Microsoft.Framework.Runtime/ApplicationHostContext.cs

```diff
@@ -60,39 +71,36 @@ public class ApplicationHostContext
             var projectLockJsonPath = Path.Combine(ProjectDirectory, LockFileReader.LockFileName);
             var lockFileExists = File.Exists(projectLockJsonPath);
             var validLockFile = false;
+            ICompilationMessage lockFileDiagnostic = null;
 
             if (lockFileExists)
             {
                 var lockFileReader = new LockFileReader();
                 var lockFile = lockFileReader.Read(projectLockJsonPath);
-                validLockFile = lockFile.IsValidForProject(Project);
 
-                if (validLockFile || skipLockFileValidation)
+                string message;
+                validLockFile = lockFile.IsValidForProject(Project, out message);
+
+                if (skipLockFileValidation || validLockFile)
                 {
                     NuGetDependencyProvider.ApplyLockFile(lockFile);
-
-                    DependencyWalker = new DependencyWalker(new IDependencyProvider[] {
-                        ProjectDepencyProvider,
-                        NuGetDependencyProvider,
-                        referenceAssemblyDependencyResolver,
-                        gacDependencyResolver,
-                        unresolvedDependencyProvider
-                    });
+                }
+                else
+                {
+                    dependencyProviders.Remove(NuGetDependencyProvider);
```

Comment:
:clap: - just add a comment why we are doing this - similarly to what we have before ("We don't add NuGetDependencyProvider to DependencyWalker
// It will leave all NuGet packages unresolved and give error message asking users to run "dnu restore")